### DISCUSSION TITLE
Allow overriding pathPrefix using a CLI argument

### DISCUSSION THREAD

**Discussion by chrispytoes** [gatsbyjs/gatsby#30725] [Ideas / Feature Requests]
I'm setting up a CI flow for a Gatsby site, and I'd like to be able to set the `pathPrefix` config value based on the environment the site is being built for. When building for development, I want to add a path prefix to host on GitLab pages, but no prefix when building for production.

Was this ever considered or is there any particular reason it hasn't been done? It seems like a no-brainer to me that this should just be a simple CLI argument like: `gatsby build --prefix-paths /my-prefix`.

I understand that the `pathPrefix` setting is ignored if `--prefix-paths` is not provided, but it's still a one or the other choice. I feel that the prefix is more of a choice made by the environment, and should not be embedded into a config setting in the first place.

**Comment by dasrecht** [comment]
This would be a very helpful way for folks that have setups that use different pathPrefixes for preview environments.

**Reply by dasrecht** [reply]
I found a workaround for this issue

```
sed -ibak "5s/blog/${PREFIX}/" gatsby-config.js
```

In the build pipeline i generate  the needed prefix and then just before building the static preview site I replace the prefix in the `gatsby-config.js` 

Is hacky but does its job

**Reply by violy** [reply]
you can also use environment variables, with `export YOUR_PREFIX_PATH=...`, `dotenv` and then `process.env.YOUR_PREFIX_PATH` inside your `gatsby-config` ?

**Reply by dasrecht** [reply]
Oh! I didn't even think that far. Yes that might also work and be less hacky :D Thanks a lot!
